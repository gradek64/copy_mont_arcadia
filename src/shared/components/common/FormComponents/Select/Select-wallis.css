/** @define Select */

.Select {
  &-select {
    padding: 0 35px 0 10px;
  }
}