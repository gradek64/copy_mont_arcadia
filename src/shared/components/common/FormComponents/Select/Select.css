/** @define Select */

.Select {
  display: flex;
  position: relative;
  width: 100%;
  margin: $input-margin-vertical 0;
  font-family: $font-family-select;
  flex-direction: column;
  letter-spacing: $select-letter-spacing;

  &-container {
    position: relative;
    color: $input-placeholder-font-color;
    &::after {
      content: '';
      position: absolute;
      right: $select-arrow-right;
      height: 100%;
      top: 0;
      width: $select-arrow-width;
      border-left: $select-arrow-border-width solid $input-pre-border-color;
      background-image: url("public/$brand/images/down-arrow.svg");
      background-repeat: no-repeat;
      background-position: $plp-filters-sort-selector-arrow-position;
      background-size: $select-arrow-size;
      box-sizing: border-box;
      pointer-events: none;
    }
  }

  &-label {
    margin-right: auto;
    margin-left: $input-label-margin-left;
    color: $input-pre-label-font-color;
    font-weight: $input-pre-label-font-weight;
    font-style: $input-pre-label-font-style;
    text-transform: $input-pre-label-text-transform;
    flex-shrink: 0;
    max-width: 100%;
  }

  &-required {
    color: $input-required-color;
  }

  &-select {
    height: $form-element-height;
    display: block;
    width: 100%;
    box-sizing: border-box;
    padding: $input-padding;
    padding-right: calc($select-arrow-width + 5px);
    font-size: $select-pre-font-size;
    letter-spacing: $select-pre-letter-spacing;
    color: $select-pre-font-color;
    background: $select-background;
    border: $input-border-width solid $select-pre-border-color;

    border-radius: $input-border-radius;
    box-shadow: $input-box-shadow;

    -moz-appearance: none;
    -webkit-appearance: none;

    &:active {
      font-weight: $input-active-font-weight;
      font-style: $input-active-font-style;
      text-transform: $input-active-text-transform;
      outline: $input-active-outline;

      color: $select-active-font-color;
      border: $input-border-width solid $select-active-border-color;
    }

    &:active + .Select-label {
      color: $input-active-label-font-color;
      font-weight: $input-active-label-font-weight;
      font-style: $input-active-label-font-style;
      text-transform: $input-active-label-text-transform;
    }

    &::-ms-expand {
      display: none;
    }
  }

  &-head {
    display: flex;
  }

  &-label, &-validationMessage {
    margin-bottom: 5px;
  }

  &-validationMessage {
    padding-left: 10px;
    color: $error-color;
    text-align: right;
  }

  &.is-selected {
    & .Select-container::after {
      border-color: $select-post-border-color;
      opacity: 1;
    }

    & .Select-select {
      color: $select-post-font-color;
      border: $input-border-width solid $select-post-border-color;

      &:-moz-focusring {
        color: transparent;
        text-shadow: 0 0 0 $select-active-font-color;
      }
    }

    & .Select-label {
      color: $input-post-label-font-color;
      font-weight: $input-post-label-font-weight;
      font-style: $input-post-label-font-style;
      text-transform: $input-post-label-text-transform;
    }
  }

  &.is-disabled {
    pointer-events: none;

    & .Select-container::after {
      border-color: $select-inactive-border-color;
      opacity: $select-inactive-arrow-opacity;
    }

    & .Select-select {
      background: $select-inactive-background;
      color: $select-inactive-font-color;
      border: $input-border-width solid $select-inactive-border-color;
    }

    & .Select-label {
      color: $input-inactive-label-font-color;
      font-weight: $input-inactive-label-font-weight;
      font-style: $input-inactive-label-font-style;
      text-transform: $input-inactive-label-text-transform;
    }
  }

  &.is-erroring {
    & .Select-container::after {
      border-color: $select-error-border-color;
      opacity: 1;
    }

    & .Select-select {
      color: $select-post-font-color;
      border-color: $select-error-border-color;
    }

    & .Select-label {
      color: $input-post-label-font-color;
    }
  }

  &--sort {
    &.is-selected .Select-select {
      border: 1px solid $plp-filters-sort-selector-border-color;
      padding: $plp-filters-sort-selector-padding;
      font-size: $plp-filters-sort-selector-font-size;
      font-family: $plp-filters-sort-selector-font-family;
      letter-spacing: $plp-filters-sort-selector-letter-spacing;
    }

    & .Select-container::after {
      width: $plp-filters-sort-selector-arrow-width;
      background-size: $plp-filters-sort-selector-arrow-background-size;
    }
  }

  &--inlineLabel {
    flex-direction: row;
    & .Select {
      &-head {
        align-self: center;
      }
      &-label {
        margin: 0 13px 0 0;
      }
      &-container {
        flex: 1;
      }
    }
  }

  &--link {
    margin: 0;
    justify-content: center;
    width: auto;

    & .Select {
      &-label {
        font-size: 0.925em;
      }
      &-container {
        &:after {
          display: none;
        }
      }
      &-select {
        appearance: none;
        height: auto;
        background: transparent;
        border: none;
        padding: 0;
        text-decoration: underline;
        text-transform: capitalize;
        white-space: normal;
        width: 142px;
      }
    }
    &.is-selected {
      & .Select-select {
        border: none;
      }
    }
  }
}

@media $min-tablet {
  .Select--sort {
    &.is-selected .Select-select {
      padding: $plp-filters-sort-selector-padding-desktop;
      font-size: $plp-filters-sort-selector-font-size-desktop;
    }

    & .Select-container::after {
      width: $plp-filters-sort-selector-arrow-width-desktop;
      background-size: $plp-filters-sort-selector-arrow-background-size-desktop;
    }
  }
}

/* postcss-bem-linter: ignore */
.Select-select--option  {
  color: $select-post-font-color;
}

/* postcss-bem-linter: ignore */
.Select-select option:disabled {
  color: $select-inactive-font-color;
}

/* IE11 Patch to hide dropdown arrows */
/* postcss-bem-linter: ignore */
.Select-select::-ms-expand {
  display: none;
}
