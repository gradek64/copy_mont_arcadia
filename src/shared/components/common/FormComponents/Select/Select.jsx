import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { connect } from 'react-redux'

import { PrivacyGuard } from '../../../../lib'
import InputError from '../InputError/InputError'
import {
  sendAnalyticsValidationState,
  GTM_VALIDATION_STATES,
} from '../../../../analytics'
import { isInCheckout as checkIfInCheckout } from '../../../../selectors/routingSelectors'

@connect(
  (state) => ({
    isInCheckout: checkIfInCheckout(state),
  }),
  { sendAnalyticsValidationState }
)
class Select extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    field: PropTypes.object,
    errors: PropTypes.object,
    options: PropTypes.array,
    firstDisabled: PropTypes.string,
    className: PropTypes.string,
    selectContainerClassName: PropTypes.string,
    label: PropTypes.string,
    hideLabel: PropTypes.bool,
    defaultValue: PropTypes.string,
    value: PropTypes.string,
    onBlur: PropTypes.func,
    isDisabled: PropTypes.bool,
    autofocus: PropTypes.bool,
    isRequired: PropTypes.bool,
    noTranslate: PropTypes.bool,
    privacyProtected: PropTypes.bool,
    isInCheckout: PropTypes.bool,
    sendAnalyticsValidationState: PropTypes.func.isRequired,
  }

  static defaultProps = {
    options: [],
    className: '',
    selectContainerClassName: '',
    label: '',
    onBlur: () => {},
    isInCheckout: false,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  // Set the displayed default value in the form model as well
  // otherwise sends empty payload if not changed
  componentDidMount() {
    if (this.props.defaultValue)
      this.changeHandler({ target: { value: this.props.defaultValue } })
    if (this.props.value) this.selected = true
  }

  // on clear form if the value changes to '' then set it to the default value
  UNSAFE_componentWillUpdate({ defaultValue, value }) {
    if (value === '' && defaultValue)
      this.changeHandler({ target: { value: defaultValue } })
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.value) this.selected = true
  }

  handleBlur = (...args) => {
    const { name, errors, onBlur, isInCheckout } = this.props
    if (errors && isInCheckout) {
      this.props.sendAnalyticsValidationState({
        id: name,
        validationStatus: errors[name]
          ? GTM_VALIDATION_STATES.FAILURE
          : GTM_VALIDATION_STATES.SUCCESS,
      })
    }
    onBlur(...args)
  }

  getSelectHead = (
    name,
    label,
    error,
    isTouched = false,
    hideLabel,
    isRequired
  ) => {
    const { l } = this.context
    const showError =
      (this.selected === true && !!error) ||
      (this.selected === undefined && !!error && isTouched)

    if (!label && !showError) return null

    return (
      <div className="Select-head">
        {label && (
          <label
            className={`Select-label ${hideLabel ? 'screen-reader-text' : ''}`}
            htmlFor={name}
          >
            {label}
            {isRequired && <span className="Select-required">*</span>}
          </label>
        )}
        {showError && (
          <InputError className="Select-validationMessage" name={name}>
            {l(error)}
          </InputError>
        )}
      </div>
    )
  }

  changeHandler = (e) => {
    const { onChange } = this.props

    onChange(e)
    this.selected = true
  }

  render() {
    const {
      className,
      label,
      options,
      name,
      field,
      value,
      firstDisabled,
      isDisabled,
      isRequired,
      errors,
      hideLabel,
      autofocus,
      noTranslate,
      privacyProtected,
      selectContainerClassName,
    } = this.props

    const containerClassName = `${name ? ` FormComponent-${name}` : ''}`

    return (
      <div
        className={`${containerClassName} Select ${className}${
          errors && errors[name] && field && field.isTouched
            ? ' is-erroring'
            : ''
        }${isDisabled ? ' is-disabled' : ''}${
          !firstDisabled || this.selected ? ' is-selected' : ''
        }`}
      >
        {this.getSelectHead(
          name,
          label,
          errors && errors[name],
          field && field.isTouched,
          hideLabel,
          isRequired
        )}
        <div className={`Select-container ${selectContainerClassName}`}>
          <PrivacyGuard noProtection={!privacyProtected}>
            <select
              autoFocus={autofocus}
              className={`Select-select`}
              name={name}
              value={value || firstDisabled}
              onChange={(e) => this.changeHandler(e)}
              onBlur={this.handleBlur}
              disabled={isDisabled}
              id={name}
              aria-label={label}
              aria-activedescendant={value}
              role="listbox"
              required={isRequired}
            >
              {firstDisabled && (
                <option
                  disabled
                  className={`Select-select--option ${
                    noTranslate ? ' notranslate' : ''
                  }`}
                >
                  {firstDisabled}
                </option>
              )}
              {options[0] instanceof Object
                ? options.map((opt, i) => (
                    <option
                      className={`Select-select--option ${
                        noTranslate ? ' notranslate' : ''
                      }`}
                      key={i} // eslint-disable-line react/no-array-index-key
                      disabled={opt.disabled}
                      value={opt.value}
                      role="option"
                      aria-selected={false}
                    >
                      {opt.label}
                    </option>
                  ))
                : options.map((opt, i) => (
                    <option
                      className={`Select-select--option ${
                        noTranslate ? ' notranslate' : ''
                      }`}
                      key={i} // eslint-disable-line react/no-array-index-key
                      value={opt}
                      role="option"
                      aria-selected={false}
                    >
                      {opt}
                    </option>
                  ))}
            </select>
          </PrivacyGuard>
        </div>
      </div>
    )
  }
}

export default Select
