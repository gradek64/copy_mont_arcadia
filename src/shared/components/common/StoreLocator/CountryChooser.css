/** @define CountryChooser */

@media $min-tablet {
  .CountryChooser {
    margin: 0;
  }
}
