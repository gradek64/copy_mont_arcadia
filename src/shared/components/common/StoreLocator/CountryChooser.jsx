import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { path } from 'ramda'

import Select from '../FormComponents/Select/Select'
import { selectCountry } from '../../../actions/components/StoreLocatorActions'
import { isFeatureYextEnabled } from '../../../../shared/selectors/featureSelectors'
import {
  getSelectedCountry,
  createOptions,
} from '../../../../shared/selectors/storeLocatorSelectors'
import { isMobile } from '../../../../shared/selectors/viewportSelectors'

@connect(
  (state) => ({
    isMobile: isMobile(state),
    selectedCountry: getSelectedCountry(state),
    isFeatureYextEnabled: isFeatureYextEnabled(state),
    createOptions: createOptions(state),
  }),
  { selectCountry }
)
class CountryChooser extends Component {
  static propTypes = {
    selectCountry: PropTypes.func.isRequired,
    className: PropTypes.string,
    isMobile: PropTypes.bool,
    name: PropTypes.string,
    selectedCountry: PropTypes.string,
    isLandingPage: PropTypes.bool,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  static defaultProps = {
    className: '',
    countries: [],
    name: 'CountrySelect',
  }

  onCountryChange = (event) => {
    const { selectCountry, selectedCountry } = this.props
    const value = path(['target', 'value'], event)
    if (selectedCountry !== value) {
      selectCountry(value)
    }
  }

  render() {
    const { l } = this.context
    const label = l`Choose country`
    const {
      className,
      isMobile,
      name,
      selectedCountry = label,
      isLandingPage,
      isFeatureYextEnabled,
      createOptions,
    } = this.props
    const yextCheck = isFeatureYextEnabled ? '' : 'Select--link'
    return (
      <Select
        id={name}
        className={`CountryChooser ${isMobile ? yextCheck : ''} ${className}`}
        name={name}
        options={createOptions}
        onChange={this.onCountryChange}
        label={isLandingPage ? l`Store Locator` : l`I'm looking for a store in`}
        hideLabel={isLandingPage}
        value={selectedCountry}
      />
    )
  }
}

export default CountryChooser
