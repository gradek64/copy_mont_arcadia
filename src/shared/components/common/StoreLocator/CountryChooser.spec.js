import testComponentHelper from 'test/unit/helpers/test-component'
import CountryChooser from './CountryChooser'
import Select from '../FormComponents/Select/Select'

describe('<CountryChooser/>', () => {
  const initialProps = {
    countries: [],
    isMobile: true,
    getCountries: jest.fn(),
    selectCountry: jest.fn(),
  }
  const countries = ['United Kingdom', 'Germany', 'Spain', 'Gibraltar']
  const renderComponent = testComponentHelper(CountryChooser.WrappedComponent)

  describe('@renders', () => {
    describe('mobile', () => {
      it('in default state', () => {
        expect(renderComponent(initialProps).getTree()).toMatchSnapshot()
      })
      it('with className', () => {
        expect(
          renderComponent({
            ...initialProps,
            className: 'test-countryChooser',
          }).getTree()
        ).toMatchSnapshot()
      })
      it('with name', () => {
        expect(
          renderComponent({
            ...initialProps,
            className: 'test-name',
          }).getTree()
        ).toMatchSnapshot()
      })
      it('with countries', () => {
        expect(
          renderComponent({
            ...initialProps,
            countries,
          }).getTree()
        ).toMatchSnapshot()
      })
      it('with selectedCountry', () => {
        expect(
          renderComponent({
            ...initialProps,
            selectedCountry: 'Gibraltar',
          }).getTree()
        ).toMatchSnapshot()
      })
      it('as landingPage', () => {
        expect(
          renderComponent({
            ...initialProps,
            isLandingPage: true,
          }).getTree()
        ).toMatchSnapshot()
      })
    })
    describe('desktop', () => {
      it('in default state', () => {
        expect(
          renderComponent({ ...initialProps, isMobile: false }).getTree()
        ).toMatchSnapshot()
      })
      it('with className', () => {
        expect(
          renderComponent({
            ...initialProps,
            isMobile: false,
            className: 'test-countryChooser',
          }).getTree()
        ).toMatchSnapshot()
      })
      it('with name', () => {
        expect(
          renderComponent({
            ...initialProps,
            isMobile: false,
            className: 'test-name',
          }).getTree()
        ).toMatchSnapshot()
      })
      it('with countries', () => {
        expect(
          renderComponent({
            ...initialProps,
            isMobile: false,
            countries,
          }).getTree()
        ).toMatchSnapshot()
      })
      it('with selectedCountry', () => {
        expect(
          renderComponent({
            ...initialProps,
            isMobile: false,
            selectedCountry: 'Gibraltar',
          }).getTree()
        ).toMatchSnapshot()
      })
      it('as landingPage', () => {
        expect(
          renderComponent({
            ...initialProps,
            isMobile: false,
            isLandingPage: true,
          }).getTree()
        ).toMatchSnapshot()
      })
    })
  })
  describe('@lifecycle', () => {
    beforeEach(() => {
      jest.resetAllMocks()
    })
  })
  describe('@events', () => {
    describe('<select />', () => {
      it('calls selectCountry handler on select', () => {
        const { wrapper, instance } = renderComponent({
          ...initialProps,
          countries,
          getCountries: jest.fn(),
          selectCountry: jest.fn(),
          selectedCountry: 'United Kingdom',
        })

        expect(instance.props.selectCountry).not.toBeCalled()
        wrapper
          .find(Select)
          .simulate('change', { target: { value: 'Germany' } })
        expect(instance.props.selectCountry).toHaveBeenCalledTimes(1)
      })

      it('does not call selectCountry when the same country have been choosen', () => {
        const { wrapper, instance } = renderComponent({
          ...initialProps,
          countries,
          getCountries: jest.fn(),
          selectCountry: jest.fn(),
          selectedCountry: 'United Kingdom',
        })

        expect(instance.props.selectCountry).not.toBeCalled()
        wrapper
          .find(Select)
          .simulate('change', { target: { value: 'United Kingdom' } })
        expect(instance.props.selectCountry).not.toBeCalled()
      })
    })
  })
})
