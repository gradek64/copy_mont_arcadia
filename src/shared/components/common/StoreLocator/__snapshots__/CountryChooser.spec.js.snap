// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryChooser/> @renders desktop as landingPage 1`] = `
<Connect(Select)
  className="CountryChooser  "
  hideLabel={true}
  id="CountrySelect"
  label="Store Locator"
  name="CountrySelect"
  onChange={[Function]}
  value="Choose country"
/>
`;

exports[`<CountryChooser/> @renders desktop in default state 1`] = `
<Connect(Select)
  className="CountryChooser  "
  id="CountrySelect"
  label="I'm looking for a store in"
  name="CountrySelect"
  onChange={[Function]}
  value="Choose country"
/>
`;

exports[`<CountryChooser/> @renders desktop with className 1`] = `
<Connect(Select)
  className="CountryChooser  test-countryChooser"
  id="CountrySelect"
  label="I'm looking for a store in"
  name="CountrySelect"
  onChange={[Function]}
  value="Choose country"
/>
`;

exports[`<CountryChooser/> @renders desktop with countries 1`] = `
<Connect(Select)
  className="CountryChooser  "
  id="CountrySelect"
  label="I'm looking for a store in"
  name="CountrySelect"
  onChange={[Function]}
  value="Choose country"
/>
`;

exports[`<CountryChooser/> @renders desktop with name 1`] = `
<Connect(Select)
  className="CountryChooser  test-name"
  id="CountrySelect"
  label="I'm looking for a store in"
  name="CountrySelect"
  onChange={[Function]}
  value="Choose country"
/>
`;

exports[`<CountryChooser/> @renders desktop with selectedCountry 1`] = `
<Connect(Select)
  className="CountryChooser  "
  id="CountrySelect"
  label="I'm looking for a store in"
  name="CountrySelect"
  onChange={[Function]}
  value="Gibraltar"
/>
`;

exports[`<CountryChooser/> @renders mobile as landingPage 1`] = `
<Connect(Select)
  className="CountryChooser Select--link "
  hideLabel={true}
  id="CountrySelect"
  label="Store Locator"
  name="CountrySelect"
  onChange={[Function]}
  value="Choose country"
/>
`;

exports[`<CountryChooser/> @renders mobile in default state 1`] = `
<Connect(Select)
  className="CountryChooser Select--link "
  id="CountrySelect"
  label="I'm looking for a store in"
  name="CountrySelect"
  onChange={[Function]}
  value="Choose country"
/>
`;

exports[`<CountryChooser/> @renders mobile with className 1`] = `
<Connect(Select)
  className="CountryChooser Select--link test-countryChooser"
  id="CountrySelect"
  label="I'm looking for a store in"
  name="CountrySelect"
  onChange={[Function]}
  value="Choose country"
/>
`;

exports[`<CountryChooser/> @renders mobile with countries 1`] = `
<Connect(Select)
  className="CountryChooser Select--link "
  id="CountrySelect"
  label="I'm looking for a store in"
  name="CountrySelect"
  onChange={[Function]}
  value="Choose country"
/>
`;

exports[`<CountryChooser/> @renders mobile with name 1`] = `
<Connect(Select)
  className="CountryChooser Select--link test-name"
  id="CountrySelect"
  label="I'm looking for a store in"
  name="CountrySelect"
  onChange={[Function]}
  value="Choose country"
/>
`;

exports[`<CountryChooser/> @renders mobile with selectedCountry 1`] = `
<Connect(Select)
  className="CountryChooser Select--link "
  id="CountrySelect"
  label="I'm looking for a store in"
  name="CountrySelect"
  onChange={[Function]}
  value="Gibraltar"
/>
`;
